src/components/modal/Modal.tsx

```diff
@@ -0,0 +1,60 @@
+import { Button, LoadingSpinner } from '@components';
+import { AiOutlineClose } from 'react-icons/ai';
+import { ModalState } from '@constants';
+import { ReactNode } from 'react';
+import styles from './Modal.module.less';
+import { useDelayedMount } from 'util/delayed-mount/delayed-mount';
+
+interface Props {
+  ariaLabel: string,
+  state: ModalState,
+  setState: (state: ModalState) => void,
+  headerString: string,
+  body: ReactNode,
+  footer?: ReactNode
```

Comment:
I think all those three should be optional ReactNode, for maximal re-usability

Reply:
did exactly that